lightautoml/ml_algo/torch_based/nn_models.py

```diff
@@ -0,0 +1,733 @@
+        """
+
+        Args:
+            stddev: Std of noise.
+            device: Device to compute on.
+
+        """
+        super().__init__()
+        self.stddev = stddev
+        self.device = device
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        if self.training:
+            return x + torch.randn(x.size(), device=self.device) * self.stddev
+        return x
+
+
+class UniformNoise(nn.Module):
+    """Add uniform noise."""
+
+    def __init__(self, stddev: float, device: torch.Tensor):
+        """
+
+        Args:
+            stddev: Std of noise.
+            device: Device to compute on.
+
+        """
+        super().__init__()
+        self.stddev = stddev
+        self.device = device
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        if self.training:
+            return x + (torch.rand(x.size(), device=self.device) - 0.5) * self.stddev
+        return x
+
+
+class DenseLightBlock(nn.Module):
+    """Realisation of `'denselight'` model block."""
+
+    def __init__(
+        self,
+        n_in: int,
+        n_out: int,
+        drop_rate: float = 0.1,
+        noise_std: float = 0.05,
+        act_fun: nn.Module = nn.ReLU,
+        use_bn: bool = True,
+        use_noise: bool = False,
+        device: torch.device = torch.device("cuda:0"),
+        **kwargs,
+    ):
+        """
+
+        Args:
+            n_in: Input dim.
+            n_out: Output dim.
+            drop_rate: Dropout rate.
+            noise_std: Std of noise.
+            act_fun: Activation function.
+            use_bn: Use BatchNorm.
+            use_noise: Use noise.
+            device: Device to compute on.
+
+        """
+
+        super(DenseLightBlock, self).__init__()
+        self.features = nn.Sequential(OrderedDict([]))
+
+        if use_bn:
+            self.features.add_module("norm", nn.BatchNorm1d(n_in))
+        if drop_rate:
+            self.features.add_module("dropout", nn.Dropout(p=drop_rate))
+        if use_noise:
+            self.features.add_module("noise", GaussianNoise(noise_std, device))
+
+        self.features.add_module("dense", nn.Linear(n_in, n_out))
+        self.features.add_module("act", act_fun())
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        for name, layer in self.features.named_children():
+            x = layer(x)
+        return x
+
+
+class DenseLightModel(nn.Module):
+    """Realisation of `'denselight'` model."""
+
+    def __init__(
+        self,
+        n_in: int,
+        n_out: int = 1,
+        hidden_size: List[int] = [
+            512,
+            750,
+        ],
+        drop_rate: Union[float, List[float]] = 0.1,
+        act_fun: nn.Module = nn.ReLU,
+        noise_std: float = 0.05,
+        num_init_features: Optional[int] = None,
+        use_bn: bool = True,
+        use_noise: bool = False,
+        concat_input: bool = True,
+        device: torch.device = torch.device("cuda:0"),
+        **kwargs,
```

Comment:
Do you need kwargs here?

Reply:
If you want to use a custom model, then params are passed to the creation method. So if we remove kwargs here, there will be no availability to use custom model (tuning your params for example).